### PR TITLE
src/Makefile: make 'make clean' clean all produced files

### PR DESCRIPTION
**Pull request by ia**
TL; DR: make `make clean` clean _all_ produced files.

Steps to reproduce:
 - run build: `$ make`
 - clean produced files: `$ make clean`

What should happen:
 - usually, it's a common practice when after `make clean` _all_ produced files are deleted

What actually happens:
 - `src/common/hydrafw_version.hdr` kept as _"leftover"_

At first I was thinking about adding a separate `clean` target into `Makefile` but since internal variables of `rules.mk` from `ChibiOS` support "custom enhancements", just adding command `rm -f src/common/hydrafw_version.hdr` to special variable do the trick.

I did successfully test it myself on `Ubuntu 18.04`. But I did not test it on `Windows`/`mingw` unfortunately.

P.S. Happy New Year! :)

### CODE REVIEW

**Review by bvernoux — APPROVED** [review]

**Comment by bvernoux** [comment]
Thanks for your contribution
